packages/api/test/user-payment.spec.js

```diff
@@ -154,11 +155,46 @@ describe('userPaymentPut', () => {
       customers.mockCustomers.map(c => c.id).includes(billing.paymentMethodSaves[0].customerId),
       'billing.paymentMethodSaves[0].customerId is in customers.mockCustomers')
   })
+  it('throws an error if no agreement', async function () {
```

Comment:
Can we add a test not for the error conditions but also to ensure that the agreement is saved?

One way of doing it would be to invoke userPaymentPut with an `env.termsOfService` that is a mock and ensure that the correct agreement value was passed to a method on that TermsOfServiceService.

(If we had a way 'from the outside' of knowing things had been saved, e.g. userPaymentGet returning a list of agreements, then we might not even need a mock, you could do userPaymentPut with an agreement and then userPaymentGet to ensure it had been saved, but eh it's not worth adding new functionality to userPaymentGet just for the test, so that's why the mocks are nice for unit testing)

Reply:
I added a lot of tests, including the unit test I described above.

I also added some tests for some various use cases of using this http api (and updated business logic accordingly), including:
* Just using the api to update the default payment method, which happens on /account/payment in lower left (but not in context of making agreements or choosing plan)
* using api to update the plan, which requires the agreement